### DISCUSSION TITLE
LDM Notes for September 30th, 2024

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#8498] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2024/LDM-2024-09-30.md

### Agenda

- Extensions

**Comment by dersia** [comment]
Regarding the `this` issue, how about supporting `this` for the underlying type and `ext` or `extension` for the extension type?

```c#
public extension C for string
{
    public int Length => 10;
    public int M()
    {
        Console.WriteLine(this.Length); // prints the length of the string
        Console.WriteLine(extension.Length); // prints 10
    }
}
```

Edit: fixed property `Length` to be non-stateful

**Reply by CyrusNajmabadi** [reply]
Extensions can't have state. For values there is no place to put them. For reference types, the only option is a conditional week table, which is far too expensive. 

If it's a non-stateful property, I don't forsee us ever preferring the extension over the instance. We made a string decision on that with methods and I don't see that changing.

**Reply by jnm2** [reply]
> conditional week table

It might be there all week, and it might not

**Reply by TahirAhmadov** [reply]
@CyrusNajmabadi where in that question did @dersia say anything about state?
PS. I see what you mean - he wrote `int Length { get; } => 10;`, but his question is still pertinent if it's `int Length => 10;`. Care to address the question at hand?

**Reply by CyrusNajmabadi** [reply]
> @CyrusNajmabadi where in that question did @dersia say anything about state?
> PS. I see what you mean - he wrote `int Length { get; } => 10;`, but his question is still pertinent if it's `int Length => 10;`. Care to address the question at hand?

I literally addressed that in my post. My post covers both non stateful and stateful properties. I was wanting to cover both since a property could be either.

**Reply by dersia** [reply]
I ment for it to be `int Length => 10`, thank for correcting @TahirAhmadov

@CyrusNajmabadi so maybe I misunderstood the meeting notes then, I thought this was exactly what was in question in those notes

**Comment by qrli** [comment]
I don't quite get the motivation to have extension type for `this`, putting roles aside. 
Old extension does not allow members to be shadowed by extensions, which I think is reasonable. Why does the new extension want to allow shadowing underlying type members?
